### PR TITLE
chore: dark mode and fixed links

### LINKED ISSUE #6870: Feature  add Hackathon module on Townsquare Experiment
### User Story

As an operator of Gitcoin, we'd like more people to register for our hacakthons.

As a sponsor of Gitcoin, I'd like great devs to join my hackathons

As a developer, I'd like to know what is available for me to do on Gitcoin.

### Why Is this Needed

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*:
We need more ways for people in townsquare to get started on hacakthons

_If we add this module, then our sponsors will be happy because more developers would register for Gitcoin hackathons because they'll know that a hackathon is going on or upcoming._

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Current Behavior
No module to drive users into hackathons

### Expected Behavior
Display in right rail of townsquare

### Definition of Done
- Build the module on townsquare, right rail above referal link
• Show 2 current hackathons when a hackathon is happening
- Show 2 upcoming hackathons if there isn't one happening.
- “Join Hackathon” brings you to each hackathon onboarding • Show max 5 hackathons
- _View all_link at the bottom that links to Hackathon List page

### Data Requirements
Track clicks on # of registrants per hackathon

### Additional Information
Screen Shot 2020-06-15 at 2 49 15 PM

### PR DESCRIPTION
**Pull request by developerfred**
##### Description
I added dark mode and new features to sign up for the hackathon

##### Refers/Fixes
#6870 #7030

##### Testing

Screen Shot 2020-07-02 at 19 47 45

### CODE REVIEW

**Comment by octavioamu** [comment]
@willsputra so the solution Fred made here is inverted the logo colors but is not good for brandings should we frame the logo into white background? 
Fred also check the card to use first the svg logo if not then fallback to png

**Comment by willsputra** [comment]
@octavioamu yea inverting the logo might not work. sponsors will complain if their logo changed colors 😂

fixed background color will be problematic too tho (what if their logo is white). could we use the same backgrounds as https://gitcoin.co/hackathon-list ?

**Comment by developerfred** [comment]
Yes @willsputra! @octavioamu I will change this.

File: app/townsquare/templates/townsquare/shared/hackathons.html [hunk 1]
```diff
@@ -1,44 +1,42 @@
...
+                        {% endif %}
+                    </div>
+                </div>
+            </div>
         </div>
-      </div>   
-    </div>
-  {% endfor %}
-  <a href="https://gitcoin.co/hackathon-list/" class="view-all">View all ></a>
+    {% endfor %}
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/shared/hackathons.html:41]: Additionally the hackathon.slug works in live! t
hose changes can be undone

File: app/townsquare/templates/townsquare/index.html [hunk 2]
```diff
@@ -197,13 +236,17 @@
               {% if SHOW_DRESSING %}
 
                 {% include 'townsquare/shared/tribe.html' %}
-               
+
 
                 {% if hackathon_tabs|length %}
                   <div class="townsquare_block-header box-collapse  mt-1" data-target="hackathons">
-                    Gitcoin Hackathons
+                    Happining Now
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/index.html:243]: spelling fix

File: app/townsquare/templates/townsquare/index.html [hunk 3]
```diff
@@ -29,11 +29,50 @@
...
+      }
+
+      .dark-mode .townsquare_block-header.box-collapse {
+        background: #373350;
+        border-bottom: 1px solid var(--gc-medium-dark-gray);
+      }
+
+      .hackathon-card-text {
+        color: #000;
+      }
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/index.html:69]: the darkmode css classes aren't needed! I think we can do without them
something like https://github.com/gitcoinco/web/pull/7062

File: app/assets/v2/css/town_square.css [hunk 4]
```diff
@@ -1366,3 +1366,23 @@ body.green.offer_view .announce {
...
+  color: rgb(59, 169, 153);
+  padding: 4px;
+
+}
+
+.tag-grants{
+  background-color: rgb(240, 240, 245);
+  color: rgb(22, 14, 93);
+  padding: 4px;
+}
```
Comment (thelostone-mc) [inline app/assets/v2/css/town_square.css:1388]: why are these added ?

File: app/townsquare/templates/townsquare/shared/upcoming.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+                      Join Hackathon >
+                  </a>
+                  {% endif %}
+              </div>
+          </div>
+      </div>
+  </div>
+  {% endfor %}
+  <a href="https://gitcoin.co/hackathon-list/" class="view-all">View more events</a>
+</div>
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/shared/upcoming.html:43]: ^ indent with 1tab == 2 spaces and also reindent files 

```
{% if hackathon.end|timesince <= "1 min" %}
   <a href="{% url 'hackathon_onboard' hackathon.slug %}">
     Join Hackathon >
   </a>
{% endif %}
```

File: app/townsquare/templates/townsquare/index.html (hunk 3 above)
Comment (developerfred) [reply app/townsquare/templates/townsquare/index.html:69]: ok Thanks @thelostone-mc!

File: app/townsquare/templates/townsquare/index.html (hunk 2 above)
Comment (developerfred) [reply app/townsquare/templates/townsquare/index.html:243]: done

File: app/assets/v2/css/town_square.css (hunk 4 above)
Comment (developerfred) [reply app/assets/v2/css/town_square.css:1388]: New design feature tags for cards

File: app/townsquare/templates/townsquare/index.html [hunk 6]
```diff
@@ -250,45 +269,48 @@
...
+								<a href="{% url 'grants_catchall_:grants' %}">
+									<p>Support public goods</p>
+									<img src="{% static 'v2/images/top-bar/grants-neg.svg' %}" alt="">
             </a>
-            <a href="{% url 'quests_index' %}">
-                <p>Learn about Web3</p>
-                <img src="{% static 'v2/images/top-bar/quests-neg.svg' %}" alt="">
+									<a href="{% url 'quests_index' %}">
+										<p>Learn about Web3</p>
+										<img src="{% static 'v2/images/top-bar/quests-neg.svg' %}" alt="">
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/index.html:293]: ^ undo indent

File: app/townsquare/templates/townsquare/index.html [hunk 7]
```diff
@@ -250,45 +269,48 @@
...
+			<script src="{% static "v2/js/lib/ipfs-api.js" %}"></script>
+			<script src="{% static "v2/js/ipfs.js" %}"></script>
+			<script src="{% static "v2/js/ethereumjs-accounts.js" %}"></script>
+			<script src="{% static "v2/js/abi.js" %}"></script>
+
+			<script src="{% static "v2/js/status.js" %}"></script>
+			<script src="{% static "v2/js/pages/townsquare.js" %}"></script>
+			<script src="{% static "v2/js/pages/join_tribe.js" %}"></script>
+	</body>
+
```
Comment (thelostone-mc) [inline app/townsquare/templates/townsquare/index.html:315]: ^ undo indent

**Comment by thelostone-mc** [comment]
@developerfred looks like indentation has changed
All html+ js + css -> 1 tab = 2 spaces
Could you undo these + re-ident the new html file with 2 spaces

**Comment by thelostone-mc** [comment]
@developerfred Making this into draft for now ! Once those changes have been addressed -> switch it to ready and we can get it in

**Comment by developerfred** [comment]
@thelostone-mc  done, I'll add the upcoming feature in another pull request

**Comment by thelostone-mc** [comment]
@developerfred We can't get this PR in until all the comments have been addressed! 
Feel free to add in new changes if you'd want but we can get PRs only after the feedback is addressed

**Comment by developerfred** [comment]
@thelostone-mc ah ok! I ended up sending to the same branch the upcoming I will remove and add in a new one.

**Comment by thelostone-mc** [comment]
Hey @developerfred we're closing this out for now cause we go the fixes in ! thanks though
